### PR TITLE
Bugfix/code coverage

### PR DESCRIPTION
**Pull request by TriangleJuice**
Changed some package versions and `tsconfig` settings in order to fix code coverage.

Encountered an extra problem: because of serious negligence of the Angular team, `@angular/compiler-cli@6.1.10` requires a lower version of typescript compared to the rest. This is fixed in Angular 7, but will give a warning in Angular 6, which you can ignore.

To test locally you can best remove the `node_modules` folder and run `npm install` again.

### CODE REVIEW

**Comment by TriangleJuice** [comment]
@jsebrech Maybe it's best that you test this on Windows too ;-)

**Comment by TriangleJuice** [comment]
@lievenvg Code coverage spoiler:
```
76.77% Statements 119/155
47.54% Branches 29/61
68.97% Functions 20/29
77.03% Lines 114/148
```

File: package.json [hunk 1]
```diff
@@ -47,43 +47,40 @@
...
-    "core-js": "^2.5.7",
-    "hoek": "^6.0.3",
+    "codelyzer": "~4.2.1",
+    "core-js": "^2.5.4",
     "jasmine-core": "~2.99.1",
-    "jasmine-spec-reporter": "~4.2.1",
-    "karma": "~3.0.0",
-    "karma-chrome-launcher": "~2.2.0",
+    "jasmine-spec-reporter": "~4.1.0",
+    "karma": "~1.7.1",
```
Comment (jsebrech) [inline package.json]: is it wise to downgrade karma to a known insecure version?

**Comment by jsebrech** [comment]
> @jsebrech Maybe it's best that you test this on Windows too ;-)

Works fine on windows

**Pull request merged**